src/toolbox/ngm-geometry-drawer.ts

```diff
@@ -531,6 +535,23 @@ export class NgmAreaOfInterestDrawer extends LitElementI18n {
     this.viewer!.scene.requestRender();
   }
 
+  async downloadVisibleGeometries() {
+    const visibleGeometries = new EntityCollection();
+    this.geometriesDataSource!.entities.values.forEach(ent => {
+      if (ent.isShowing) {
+        visibleGeometries.add(ent);
+      }
+    });
+    const exportResult: exportKmlResultKml = <exportKmlResultKml> await exportKml({
+      entities: visibleGeometries,
+      time: this.julianDate
+    });
+    let kml: string = exportResult.kml;
+    kml = extendKmlWithProperties(kml, visibleGeometries);
+    const blob = new Blob([kml], {type: 'text/xml'});
```

Comment:
See the correct mime/type here: https://developers.google.com/kml/documentation/kml_tut#kml_server